src/Translator.js

```diff
@@ -90,8 +90,7 @@ class Translator extends EventEmitter {
       /* eslint prefer-rest-params: 0 */
       opt = this.options.overloadTranslationOptionHandler(arguments);
     }
-    if (typeof options === 'object') opt = { ...opt };
-    if (!opt) opt = {};
```

Comment:
If the problem is that the keySeparator isn't being propagated to `keysFromSelector`, I think there's a less risky way to fix it.

This is the only line that changed in this file when we introduced the selector API:

```javascript
// line 98
if (typeof keys === 'function') keys = keysFromSelector(keys, opt);
```

@vchirikov can you test that this also fixes the issue?

```javascript
// line 98
if (typeof keys === 'function') keys = keysFromSelector(keys, { ...this.options, ...opt });
```